src/magic_ter_fur_transform.cpp

```diff
@@ -0,0 +1,287 @@
+#include "magic.h"
+
+#include "creature.h"
+#include "point.h"
+#include "game.h"
+#include "generic_factory.h"
+#include "map.h"
+#include "mapdata.h"
+#include "messages.h"
+#include "type_id.h"
+
+namespace
+{
+generic_factory<ter_furn_transform> ter_furn_transform_factory( "ter_furn_transform" );
+} // namespace
+
+template<>
+const ter_furn_transform &string_id<ter_furn_transform>::obj() const
+{
+    return ter_furn_transform_factory.obj( *this );
+}
+
+template<>
+bool string_id<ter_furn_transform>::is_valid() const
+{
+    return ter_furn_transform_factory.is_valid( *this );
+}
+
+void ter_furn_transform::load_transform( JsonObject &jo, const std::string &src )
+{
+    ter_furn_transform_factory.load( jo, src );
+}
+
+void ter_furn_transform::reset_all()
+{
+    ter_furn_transform_factory.reset();
+}
+
+bool ter_furn_transform::is_valid() const
+{
+    return ter_furn_transform_factory.is_valid( this->id );
+}
+
+void ter_furn_transform::check_consistency()
```

Comment:
If you don't implement anything here, you might as well not define the function. What's the point of it now?